---
output: github_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning = FALSE, message = FALSE, options(readr.show_col_types = FALSE))
```

# Alignments of sequences which contribute to high Figure 1 peaks in Cases

```{r}
library(tidyverse)
library(ampir)
library(fuzzyjoin)
library(ggmsa)
library(Biostrings)
source("scripts/read_ev_polyprotein_uniprot_metadata.R")
source("scripts/get_peak_coordinates.R")
source("scripts/get_cases_peak_sequences.R")
```

### Export CXVB1 VP1 and 3D sequences for MSA

```{r}
coxsackievirusB1_P08291 <- read_ev_polyprotein_uniprot_metadata("raw_data/coxsackievirusB1_P08291.tsv")

if(file.exists("cache/sp_P08291_VP1.fasta") &&
   file.exists("cache/sp_P08291_3D.fasta")){
  message("Protein FASTA files for the VP1 and 3D region exist")
} else {
  
  source("scripts/export_ev_prot_seq.R")
  
export_ev_prot_seq(coxsackievirusB1_P08291, ev_protein = "VP1", ev_protein_suffix = "sp_P08291_")
export_ev_prot_seq(coxsackievirusB1_P08291, ev_protein = "3D", ev_protein_suffix = "sp_P08291_")
}
```

Read in data prepared in `01_figure_01_CXVB_antigen_mapping.Rmd`

```{r}
endia_virscan_onset <- read_rds("cache/endia_virscan_metadata.rds") %>% filter(onset_visit == 1)
endia_ms <- read_rds("cache/endia_ms.rds")
vigr_virscan_metadata <- read_rds("cache/vigr_virscan_metadata.rds")
vigr_ms <- read_rds("cache/vigr_ms.rds")
```

### ENDIA VP1 and 3D Peak locations

Joining the moving sum dataframe to the polyprotein genome dataframe (coxsackievirusB1_P08291) with `fuzzy_left_join` to match which proteins the peptides match to.
Note this only works if a peptide is fully aligned to the protein, if a peptide e.g., located across 2 proteins, its protein location becomes `NA`

```{r}
endia_ms_w_polyprot <- endia_ms %>%
  fuzzy_left_join(coxsackievirusB1_P08291, by = c("start", "end"),
                  match_fun = list(`>=`, `<=`)) %>% 
  drop_na()

get_peak_coordinates(ms_w_pprot_df = endia_ms_w_polyprot,
                     ev_prot_name = "VP1",
                     ev_prot_start_col = start.y,
                     condition = "cases")
```

- `start`: 1 = `window_start`: 572
- `end`: 14 = `window_start`: 585

```{r}
endia_3d_cases_peak <- get_peak_coordinates(endia_ms_w_polyprot,
                     "3D",
                     start.y) 

endia_3d_cases_peak %>% dplyr::slice(13:34)
```

- `moving_sum` > 1000 
- `start`: 14 = `window_start`: 1735
- `end`: 33 = `window_start`: 1754


Get the sequences for the peaks in cases and export them as FASTA file 
```{r}
get_cases_peak_sequences(virscan_df = endia_virscan_onset, ms_w_pprot_df = endia_ms_w_polyprot, ev_prot_name = "VP1") %>% 
  df_to_faa(file = "cache/endia_cxvb1_VP1_case_pep.fasta")

get_cases_peak_sequences(virscan_df = endia_virscan_onset, ms_w_pprot_df = endia_ms_w_polyprot, ev_prot_name = "3D") %>% 
  df_to_faa(file = "cache/endia_cxvb1_3D_case_pep.fasta")
```

### VIGR

Find the location of the VP1 and 3D peaks by examining the start and end coordinates and moving sum value

```{r}
vigr_ms_w_polyprot <- vigr_ms %>%
  fuzzy_left_join(coxsackievirusB1_P08291, by = c("start", "end"),
                  match_fun = list(`>=`, `<=`)) %>% 
  drop_na()

vigr_vp1_cases_peak <- get_peak_coordinates(ms_w_pprot_df = vigr_ms_w_polyprot,
                     ev_prot_name = "VP1",
                     ev_prot_start_col = start.y,
                     condition = "cases") 

vigr_vp1_cases_peak %>% dplyr::slice(2:21)
```

- `moving_sum` > 300 (matching ENDIA)
- `start`: 25 = `window_start`: 596
- `end`: 42 = `window_start`: 613

3D region in cases has many more higher `moving_sum` values compared to the VP1 region which makes the peak regions much too long. Therefore, a threshold of `moving_sum > 1000` was used to extract peak values. (similar to ENDIA)

```{r}
vigr_3d_cases_peak <- get_peak_coordinates(vigr_ms_w_polyprot,
                     "3D",
                     start.y) 

vigr_3d_cases_peak %>% dplyr::slice(15:34)
```

- `moving_sum` > 1000 (matching ENDIA)
- `start`: 16 = `window_start`: 1737
- `end`: 33 = `window_start`: 1754

Get the sequences for the peaks in cases and export them as FASTA file 
```{r}
get_cases_peak_sequences(virscan_df = vigr_virscan_metadata, ms_w_pprot_df = vigr_ms_w_polyprot, ev_prot_name = "VP1") %>% 
  df_to_faa(file = "cache/vigr_cxvb1_VP1_case_pep.fasta")

get_cases_peak_sequences(virscan_df = vigr_virscan_metadata, ms_w_pprot_df = vigr_ms_w_polyprot, ev_prot_name = "3D") %>% 
  df_to_faa(file = "cache/vigr_cxvb1_3D_case_pep.fasta")
```

## MSA 

MSA was performed with [MAFFT version 7 webserver](https://mafft.cbrc.jp/alignment/server/add_sarscov2.html?mar15) using CLUSTAL format alignment by MAFFT (v7.511) with the Keep Alignment Length option [Katoh, Rozewicki and Yamada 2017](https://doi.org/10.1093/bib/bbx108). Alignments were then sorted in MSA viewer with Sequences down and Consensus sequence to top [Yachdav et al. 2016](https://doi.org/10.1093/bioinformatics/btw474).

ENDIA VP1 peak for cases

```{r}
endia_case_vp1_msa <- readAAStringSet("raw_data/msa_results/endia_case_vp1.fasta")

ggmsa(endia_case_vp1_msa[1:4], start = 1, end = 14, char_width = 0.5, seq_name = FALSE, color = "Hydrophobicity", consensus_views = TRUE, disagreement = FALSE, ref = "sp_P08291_VP1", border = "white") +
    geom_seqlogo(color = "Chemistry_AA", adaptive = TRUE)
```

```{r, echo = FALSE}
ggsave("figures/endia_VP1_cases1-14.svg", dpi = 400, width = 8, height = 8)
```

ENDIA 3D peak for cases

```{r}
endia_case_3d_msa <- readAAStringSet("raw_data/msa_results/endia_case_3d.fasta")

ggmsa(endia_case_3d_msa[1:11], start = 14, end = 33, char_width = 0.5, seq_name = FALSE, color = "Hydrophobicity", consensus_views = TRUE, disagreement = FALSE, ref = "sp_P08291_3D", border = "white") +
    geom_seqlogo(color = "Chemistry_AA", adaptive = TRUE)
```

```{r, echo = FALSE}
ggsave("figures/endia_3D_cases14-33.svg", dpi = 400, width = 8, height = 8)
```

VIGR VP1 for cases

```{r}
vigr_case_vp1_msa <- readAAStringSet("raw_data/msa_results/vigr_case_vp1.fasta")

ggmsa(vigr_case_vp1_msa[1:11], start = 25, end = 42, char_width = 0.5, seq_name = FALSE, color = "Hydrophobicity", consensus_views = TRUE, disagreement = FALSE, ref = "sp_P08291_VP1", border = "white") +
    geom_seqlogo(color = "Chemistry_AA", adaptive = TRUE)
```

```{r, echo = FALSE}
ggsave("figures/vigr_VP1_cases25-42.svg", dpi = 400, width = 8, height = 8)
```

VIGR 3D peak for cases

```{r}
vigr_case_3d_msa <- readAAStringSet("raw_data/msa_results/vigr_case_3d.fasta")

vigr_case_3d_msa_tidy <- c(vigr_case_3d_msa[1:4], vigr_case_3d_msa[6:7], vigr_case_3d_msa[9:10], vigr_case_3d_msa[12:13])

ggmsa(vigr_case_3d_msa_tidy, start = 16, end = 33, char_width = 0.5, seq_name = FALSE, color = "Hydrophobicity", consensus_views = TRUE, disagreement = FALSE, ref = "sp_P08291_3D", border = "white") +
    geom_seqlogo(color = "Chemistry_AA", adaptive = TRUE)
```

```{r, echo = FALSE}
ggsave("figures/vigr_3D_cases16-33.svg", dpi = 400, width = 8, height = 8)
```

:construction: trying to find a way how to sort them without using webservers

Trying it with command line instead

```bash
mafft --auto --reorder --keeplength --anysymbol --maxambiguous 0.05 --addfragments cache/endia_cxvb1_VP1_case_pep.fasta cache/sp_P08291_VP1.fasta > test.fasta 

mafft --auto --keeplength --anysymbol --maxambiguous 0.05 --addfragments cache/endia_cxvb1_VP1_case_pep.fasta cache/sp_P08291_VP1.fasta > test2.fasta 
```

```{r, eval = FALSE}
test <- readAAStringSet("test.fasta")

sorted_sequences_test <- test[order(as.character(test), decreasing = TRUE)]

ggmsa(test, start = 9, end = 14, char_width = 0.5, seq_name = FALSE, color = "Hydrophobicity", consensus_views = TRUE, disagreement = FALSE, ref = "sp|P08291|VP1", border = "white") +
    geom_seqlogo(color = "Chemistry_AA", adaptive = TRUE)

ggmsa(sorted_sequences_test[1:4], start = 9, end = 14, char_width = 0.5, seq_name = FALSE, color = "Hydrophobicity", consensus_views = TRUE, disagreement = FALSE, ref = "sp|P08291|VP1", border = "white") +
    geom_seqlogo(color = "Chemistry_AA", adaptive = TRUE)
```