---
output: github_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning = FALSE, message = FALSE, options(readr.show_col_types = FALSE))
```

# Statistical testing of antigen levels against motifs in *Enterovirus* VP1 and 3D region peaks

```{r}
library(tidyverse)
source("scripts/read_blast.R")
library(glmmTMB)
library(gt)
library(patchwork)
source("scripts/drop_unused_factor_levels.R")
```

Read in data prepared in `00_prepare_datasets.Rmd` 
and add normalised counts measured in RPK and library size log variable to use as an offset for sequencing depth

```{r}
endia_virscan_onset <- read_rds("cache/endia_virscan_metadata.rds") %>%
  filter(onset_visit == 1) %>% 
  group_by(sample_id) %>% 
  mutate(rpk = abundance / sum(abundance) * 100000,
         log_lib = log(sum(abundance) + 1e-8)) %>%  # add offset 
  ungroup()

vigr_virscan_metadata <- read_rds("cache/vigr_virscan_metadata.rds") %>% 
  group_by(sample_id) %>% 
  mutate(rpk = abundance / sum(abundance) * 100000,
         log_lib = log(sum(abundance) + 1e-8)) %>% 
  ungroup()
```

```bash
makeblastdb -in raw_data/conserved_ev_motifs.fasta -dbtype prot -out raw_data/blast_databases/conserved_ev_motifs_db

blastp -task blastp-short -query cache/endia_virscan_hits_peptides.fasta -db raw_data/blast_databases/conserved_ev_motifs_db -outfmt '6 qaccver saccver pident nident length evalue bitscore mismatch gapopen qstart qend sstart send qseq sseq ppos stitle frames' -evalue 0.01 -word_size 2 -out raw_data/blast_results/blastp_endia_ev_motifs.blast

blastp -task blastp-short -query cache/vigr_virscan_hits_peptides.fasta -db raw_data/blast_databases/conserved_ev_motifs_db -outfmt '6 qaccver saccver pident nident length evalue bitscore mismatch gapopen qstart qend sstart send qseq sseq ppos stitle frames' -evalue 0.01 -word_size 2 -out raw_data/blast_results/blastp_vigr_ev_motifs.blast
```

```{r}
endia_blastp_ev_motifs <- read_blast("raw_data/blast_results/blastp_endia_ev_motifs.blast")

vigr_blastp_ev_motifs <- read_blast("raw_data/blast_results/blastp_vigr_ev_motifs.blast")
```

```{r}
endia_motif_results <- endia_blastp_ev_motifs %>%
    left_join(endia_virscan_onset, by = join_by(qaccver == pep_id)) 

vigr_motif_results <- vigr_blastp_ev_motifs %>% 
   left_join(vigr_virscan_metadata, by = join_by(qaccver == pep_id))
```

## ENDIA

```{r}
endia_motif_w_genera <- endia_motif_results %>%
  group_by(saccver, condition, taxon_genus) %>% 
  summarise(total_rpk = sum(rpk),
            no_of_peptides = n_distinct(qaccver),
            no_of_participants = n_distinct(participant_id),
            no_of_samples = n_distinct(sample_id),
            .groups = "drop") %>% 
  filter(total_rpk > 0)
```

```{r, fig.height=8}
endia_motif_w_genera %>%
  filter(taxon_genus == "Enterovirus") %>% 
  ggplot(aes(x = taxon_genus, y = total_rpk, fill = condition)) +
  geom_col(position = position_dodge(width = 0.9)) +
  scale_fill_manual(values = c("Case" = "#d73027", "Control" = "#4575b4"), labels = c("Case", "Control")) +
  geom_text(aes(label = no_of_peptides), 
            position = position_dodge(width = 0.9), 
            vjust = -0.2, hjust = -0.7, size = 3) +
  geom_text(aes(label = no_of_samples), 
            position = position_dodge(width = 0.9), 
            vjust = -2, hjust = -0.7, size = 3, color = "purple") +
  facet_wrap(~ saccver) +
  theme_minimal(base_size = 14) +
  theme(legend.position = "right",
        line = element_blank(),
      axis.text.x = element_blank(), 
      strip.text = element_text(face = "bold")) +
  labs(x = "Enterovirus peptides with significant hits", y = "Total normalised abundance", fill = "")
```

This essentially shows that the same peptides (black numbers) are present in all samples (purple number) between case and control

```{r, echo = FALSE, fig.height=8}
ggsave("figures/ev_hits_to_motifs.png", width = 8, height = 7)
```

Having a look at the BLAST results for the VP1 motif in ENDIA, stratified by sex:
Female controls appear to have a stronger reaction compared to male controls. However, male cases have a stronger reaction compared to female cases. Black numbers are number of peptides and purple number are number of samples

```{r, echo = FALSE, fig.height=7}
endia_motif_results %>%
  group_by(saccver, condition, taxon_genus, infant_sex) %>% 
  summarise(total_rpk = sum(rpk),
            no_of_peptides = n_distinct(qaccver),
            no_of_participants = n_distinct(participant_id),
            no_of_samples = n_distinct(sample_id),
            .groups = "drop") %>% 
  filter(total_rpk > 0) %>% 
  filter(taxon_genus == "Enterovirus") %>% 
  filter(saccver == "VP1_motif") %>% 
  ggplot(aes(x = taxon_genus, y = total_rpk, fill = condition)) +
  geom_col(position = position_dodge(width = 0.9)) +
  scale_fill_manual(values = c("Case" = "#d73027", "Control" = "#4575b4"), labels = c("Case", "Control")) +
  geom_text(aes(label = no_of_peptides), 
            position = position_dodge(width = 0.9), 
            vjust = -0.2, hjust = -0.7, size = 3) +
  geom_text(aes(label = no_of_samples), 
            position = position_dodge(width = 0.9), 
            vjust = -2, hjust = -0.7, size = 3, color = "purple") +
  facet_wrap(~ infant_sex) +
  theme_minimal(base_size = 14) +
  theme(legend.position = "right",
        line = element_blank(),
      axis.text.x = element_blank(), 
      strip.text = element_text(face = "bold")) +
  labs(x = "Enterovirus peptides with significant hits to VP1 motif", y = "Total normalised abundance", fill = "")
```

```{r}
tidy_looksie <- function(glm_model) {
    glm_model %>% 
    broom.mixed::tidy(effect = "fixed", conf.int = TRUE, exponentiate = TRUE) %>% 
    select(term, estimate, p.value, conf.low, conf.high) %>% 
    mutate(significant = ifelse(p.value < 0.1, T, F))}
```

### Stats ENDIA

```{r}
endia_motif_for_stats <- endia_motif_results %>% 
  filter(taxon_genus == "Enterovirus") %>% 
  group_by(saccver, sample_id) %>% 
  mutate(total_rpk = sum(rpk),
         total_abundance = sum(abundance)) %>% 
  ungroup() %>% 
  select(saccver, participant_id, sample_id, case, condition, taxon_genus, total_rpk, infant_id, mother_id, 
         deidentified_nest_id_new, age_at_sample_collection_days, infant_sex,
         infant_HLA, weightTEDDY, maternal_T1D, total_abundance, log_lib) %>% 
  distinct(saccver, taxon_genus, sample_id, .keep_all = TRUE) %>% 
  mutate(age_sample_collection_month = age_at_sample_collection_days / 365 * 12) %>% 
  mutate(log_total_rpk = log(total_rpk + 1)) %>% 
  mutate(across(c(infant_HLA, maternal_T1D, mother_id, deidentified_nest_id_new, infant_sex), as.factor)) %>% 
  mutate(Condition = factor(condition, levels = c("Control", "Case"), labels = c("Control", "Case"))) %>% 
  mutate(infant_HLA = fct_relevel(infant_HLA, "DRXX"))
```

#### ENDIA 3D region 

```{r}
endia_3d_motif_for_stats <- endia_motif_for_stats %>% 
  filter(saccver == "3D_motif") %>% 
  drop_unused_factor_levels()

endia_3d_model <- glmmTMB(formula = case ~ log_total_rpk + infant_HLA + age_sample_collection_month + infant_sex + maternal_T1D + (1|deidentified_nest_id_new) + (1|mother_id),
       family = poisson(link = "log"),
        weights = weightTEDDY,
       data = endia_3d_motif_for_stats)

endia_3d_model %>% tidy_looksie()
```

##### 3D stratified by sex 

For the motif analyses, we had to remove the two control samples that seroconverted to cases (from 99 to 97 control samples), as if we kept these samples in, we had to include `infant_id` as random effect to account for that participant having multiple samples. However, likely due to the reduced dataset, the motif models were unable to deal with this extra layer of complexity and could not converge.

Results show no significance in abundances but HLA DR34 is significant in females 

```{r}
endia_3d_motif_for_stats_xy <- endia_3d_motif_for_stats %>% 
  filter(infant_sex == "Male") %>% 
  drop_unused_factor_levels() 

endia_3d_motif_for_stats_xx <- endia_3d_motif_for_stats %>%
  filter(infant_sex == "Female") %>% 
  drop_unused_factor_levels() 

endia_3d_model_xy <- glmmTMB(formula = case ~ log_total_rpk + infant_HLA + age_sample_collection_month + maternal_T1D + (1|deidentified_nest_id_new) + (1|mother_id),
       family = poisson(link = "log"),
        weights = weightTEDDY,
       data = endia_3d_motif_for_stats_xy)

endia_3d_model_xx <- glmmTMB(formula = case ~ log_total_rpk + infant_HLA + age_sample_collection_month + maternal_T1D + (1|deidentified_nest_id_new) + (1|mother_id),
       family = poisson(link = "log"),
        weights = weightTEDDY,
       data = endia_3d_motif_for_stats_xx)

endia_3d_model_xy %>% tidy_looksie()

endia_3d_model_xx %>% tidy_looksie()
```

3D does not appear to be a good predictor for IA. However, as observed in [Figure 1](https://github.com/TheWookHub/enterovirus-antigens-kids-w-IA/blob/main/01_figure_01_CXVB_antigen_mapping_files/figure-gfm/unnamed-chunk-11-1.png), 3D appears enriched in cases, consistent in both cohorts and [both sexes](https://github.com/TheWookHub/enterovirus-antigens-kids-w-IA/blob/main/03_figure_02B_sex_stratification_files/figure-gfm/unnamed-chunk-7-1.png)

Therefore, a statistical test was performed with the total counts for the 3D motif as outcome variable.

The distribution of the total counts for the 3D region in ENDIA. Could fit a negative binomial or poisson

```{r}
endia_3d_total_counts_distribution_plot <- endia_3d_motif_for_stats %>% 
  ggplot(aes(x = total_abundance)) +
  geom_histogram(bins = 40, fill = "black") +
  theme_minimal() +
  labs(x = "Total counts", y = "Frequency")

endia_3d_total_counts_distribution_plot
```

Distribution could be poisson (with zero inf) or neg binom. 
All three tested with anova and NB came out best.

```{r}
endia_3d_abundance_as_outcome_orig_poisson <- glmmTMB(formula = total_abundance ~ Condition + infant_HLA + age_sample_collection_month + infant_sex + maternal_T1D + offset(log_lib) + (1|deidentified_nest_id_new) + (1|mother_id),
       family = poisson(link = "log"),
        weights = weightTEDDY,
       data = endia_3d_motif_for_stats) 

endia_3d_abundance_as_outcome_poisson_with_zero <- glmmTMB(formula = total_abundance ~ Condition + infant_HLA + age_sample_collection_month + infant_sex + maternal_T1D + offset(log_lib) + (1|deidentified_nest_id_new) + (1|mother_id),
       family = poisson(link = "log"),
        weights = weightTEDDY,
       zi = ~1,
       control = glmmTMBControl(optimizer = optim, optArgs = list(method = "BFGS")),
       data = endia_3d_motif_for_stats) 

library(performance)

check_overdispersion(endia_3d_abundance_as_outcome_orig_poisson)

endia_3d_abundance_as_outcome_nb <- glmmTMB(formula = total_abundance ~ Condition + infant_HLA + age_sample_collection_month + infant_sex + maternal_T1D + offset(log_lib) + (1|deidentified_nest_id_new) + (1|mother_id),
       family = nbinom1(link = "log"),
        weights = weightTEDDY,
       data = endia_3d_motif_for_stats)

anova(endia_3d_abundance_as_outcome_orig_poisson,
      endia_3d_abundance_as_outcome_poisson_with_zero,
      endia_3d_abundance_as_outcome_nb)
```

This enrichment appears in a statistical test too:

```{r}
endia_3d_abundance_as_outcome <- endia_3d_abundance_as_outcome_nb 

endia_3d_abundance_as_outcome %>% summary()
```

#### VP1 

No significance in log_total_rpk but HLA DR34 is significant 

```{r}
endia_vp1_motif_for_stats <- endia_motif_for_stats %>% 
  filter(saccver == "VP1_motif") %>% 
  drop_unused_factor_levels()

endia_vp1_model <- glmmTMB(formula = case ~ log_total_rpk + infant_HLA + age_sample_collection_month + infant_sex + maternal_T1D + (1|deidentified_nest_id_new) + (1|mother_id),
       family = poisson(link = "log"),
        weights = weightTEDDY,
       data = endia_vp1_motif_for_stats)

endia_vp1_model %>% tidy_looksie()
```

##### VP1 stratified by sex ENDIA

Stratifying by sex results in significant difference in abundance (`log_total_rpk`) for males, but not for females. 
However, HLA DR34 is significant in females, but not for males, similar to the 3D results stratified by sex.

Number of males and females per case or control group:
```{r}
endia_vp1_motif_for_stats %>% 
  count(condition, infant_sex, name = "count") %>% 
  arrange(infant_sex)
```

```{r}
endia_vp1_motif_for_stats_xy <- endia_vp1_motif_for_stats %>% 
  filter(infant_sex == "Male") %>% 
  drop_unused_factor_levels()

endia_vp1_motif_for_stats_xx <- endia_vp1_motif_for_stats %>%
  filter(infant_sex == "Female") %>% 
  drop_unused_factor_levels()

endia_vp1_xy_model <- glmmTMB(formula = case ~ log_total_rpk + infant_HLA + age_sample_collection_month + maternal_T1D +  (1|deidentified_nest_id_new) + (1|mother_id),
       family = poisson(link = "log"),
        weights = weightTEDDY,
       data = endia_vp1_motif_for_stats_xy)

endia_vp1_xx_model <- glmmTMB(formula = case ~ log_total_rpk + infant_HLA + age_sample_collection_month + maternal_T1D + (1|deidentified_nest_id_new) + (1|mother_id),
       family = poisson(link = "log"),
        weights = weightTEDDY,
       data = endia_vp1_motif_for_stats_xx)

summary(endia_vp1_xy_model)
summary(endia_vp1_xx_model)
```

No significance for sex in an interaction model (log_total_rpk * infant_sex)
```{r}
endia_vp1_model_sex_inter <- glmmTMB(formula = case ~ log_total_rpk * infant_sex + infant_HLA + age_sample_collection_month + maternal_T1D + (1|deidentified_nest_id_new) + (1|mother_id),
       family = poisson(link = "log"),
        weights = weightTEDDY,
       data = endia_vp1_motif_for_stats)

endia_vp1_model_sex_inter %>% tidy_looksie()
```

## VIGR

```{r}
vigr_motif_w_genera <- vigr_motif_results %>%
  group_by(saccver, Condition, taxon_genus) %>% 
  summarise(total_rpk = sum(rpk),
            no_of_peptides = n_distinct(qaccver),
            no_of_participants = n_distinct(participant_id),
            no_of_samples = n_distinct(sample_id),
            .groups = "drop") %>% 
  filter(total_rpk > 0)
```

Examining BLAST results shows that in VIGR, the same peptides (black numbers) are present in all participants (purple number) between case and control

```{r, fig.height=8}
vigr_motif_w_genera %>%
  filter(taxon_genus == "Enterovirus") %>% 
  ggplot(aes(x = taxon_genus, y = total_rpk, fill = Condition)) +
  geom_col(position = position_dodge(width = 0.9)) +
  scale_fill_manual(values = c("Case" = "#d73027", "Control" = "#4575b4"), labels = c("Case", "Control")) +
  geom_text(aes(label = no_of_peptides), 
            position = position_dodge(width = 0.9), 
            vjust = -0.2, hjust = -0.7, size = 3) +
  geom_text(aes(label = no_of_samples), 
            position = position_dodge(width = 0.9), 
            vjust = -2, hjust = -0.7, size = 3, color = "purple") +
  facet_wrap(~ saccver) +
  theme_minimal(base_size = 14) +
  theme(legend.position = "right",
        line = element_blank(),
      axis.text.x = element_blank(), 
      strip.text = element_text(face = "bold")) +
  labs(x = "Enterovirus peptides with significant hits", y = "Total normalised abundance", fill = "")
```

### Stats VIGR 

No significance for VIGR in any fixed effects for any models. However, the total sample size for VIGR is small (42). Furthemore, after stratifying by sex, only ~10 males or females remain in both case and control conditions, which therefore limits the statistical power. 

```{r}
vigr_motif_for_stats <- vigr_motif_results %>% 
  mutate(Nest = str_extract(sample_id, "\\d+")) %>% 
  filter(taxon_genus == "Enterovirus") %>% 
  group_by(saccver, sample_id) %>%
  mutate(total_rpk = sum(rpk),
         total_abundance = sum(abundance)) %>% 
  ungroup() %>% 
  mutate(maternal_T1D = ifelse(maternal_T1D == "T1D", 1, 0)) %>% # encode to match ENDIA
  select(saccver, Condition, total_rpk, sample_id, taxon_genus, 
         Nest, Age, Sex, HLA_Status, My_name, total_abundance, log_lib, maternal_T1D) %>% 
 distinct(saccver, taxon_genus, sample_id, .keep_all = TRUE) %>% 
 mutate(case = ifelse(Condition == "Case", 1, 0),
        log_total_rpk = log(total_rpk + 1),
        across(c(HLA_Status, Sex, Nest, My_name, maternal_T1D), as.factor)) %>% 
 mutate(Condition = factor(Condition, levels = c("Control", "Case"), labels = c("Control", "Case")))
```

Number of males and females per case or control group:
```{r}
vigr_motif_for_stats %>% 
  distinct(sample_id, .keep_all = TRUE) %>% 
  count(Condition, Sex, name = "count")
```

#### VIGR 3D region

```{r}
vigr_3d_motif_for_stats <- vigr_motif_for_stats %>% 
  filter(saccver == "3D_motif") %>% 
  drop_unused_factor_levels()

vigr_3d_model <- glmmTMB(formula = case ~ log_total_rpk + HLA_Status + Age + Sex + maternal_T1D + (1|Nest),
       family = poisson(link = "log"),
       data = vigr_3d_motif_for_stats)

vigr_3d_model %>% tidy_looksie()
```

```{r}
vigr_3d_total_counts_distribution_plot <- vigr_3d_motif_for_stats %>% 
  ggplot(aes(x = total_abundance)) +
  geom_histogram(bins = 40, fill = "black") +
  theme_minimal() +
  labs(x = "Total counts", y = "Frequency")

vigr_3d_total_counts_distribution_plot
```

```{r}
vigr_3d_abundance_as_outcome <- glmmTMB(formula = total_abundance ~ Condition + HLA_Status + Age + Sex + maternal_T1D + offset(log_lib) + (1|Nest),
       family = nbinom1(link = "log"),
       data = vigr_3d_motif_for_stats)  

vigr_3d_abundance_as_outcome %>% tidy_looksie()
```

#### VIGR VP1 region

```{r}
vigr_vp1_motif_for_stats <- vigr_motif_for_stats %>% 
  filter(saccver == "VP1_motif") %>% 
  drop_unused_factor_levels()

vigr_vp1_model <- glmmTMB(formula = case ~ log_total_rpk + HLA_Status + Age + Sex + maternal_T1D + (1|Nest),
       family = poisson(link = "log"),
       data = vigr_vp1_motif_for_stats)

vigr_vp1_model %>% tidy_looksie()
```

#### VIGR VP1 Sex stratified 

Females VP1 motif
```{r}
vigr_vp1_motif_for_stats_xx <- vigr_vp1_motif_for_stats %>% 
  filter(Sex == "F") %>% 
  drop_unused_factor_levels()

vigr_vp1_xx_model <- glmmTMB(formula = case ~ log_total_rpk + HLA_Status + Age + maternal_T1D + (1|Nest),
       family = poisson(link = "log"),
       data = vigr_vp1_motif_for_stats_xx)

vigr_vp1_xx_model %>% tidy_looksie()
```

Males VP1 motif

```{r}
vigr_vp1_motif_for_stats_xy <- vigr_vp1_motif_for_stats %>% 
  filter(Sex == "M") %>% 
  drop_unused_factor_levels()

vigr_vp1_xy_model <- glmmTMB(formula = case ~ log_total_rpk + HLA_Status + Age + maternal_T1D + (1|Nest),
       family = poisson(link = "log"),
       data = vigr_vp1_motif_for_stats_xy)

vigr_vp1_xy_model %>% tidy_looksie()
```

### Generating pretty tables with the `gt` package from the GLM model results for the paper

#### VP1 Sex stratified 

**ENDIA**

```{r, echo = FALSE}
endia_vp1_xx_model_tidy <- endia_vp1_xx_model %>% 
  broom.mixed::tidy(effect = "fixed", conf.int = TRUE, exponentiate = TRUE) %>% 
    add_column(sex = "Females")

endia_vp1_xy_model_tidy <- endia_vp1_xy_model %>% 
  broom.mixed::tidy(effect = "fixed", conf.int = TRUE, exponentiate = TRUE) %>% 
    add_column(sex = "Males")

endia_vp1_sex_s_models_tidy <- rbind(endia_vp1_xx_model_tidy, endia_vp1_xy_model_tidy) %>% 
    dplyr::rename(RR = estimate,
               `CI lower` = conf.low,
               `CI upper` = conf.high) %>% 
    select(term, RR, `CI lower`, `CI upper`, p.value, sex) %>% 
    filter(term != "(Intercept)")

endia_vp1_sex_s_models_tidy_table <- endia_vp1_sex_s_models_tidy %>%
  mutate(term = case_match(term, !!!c("log_total_rpk" ~ "log(EV)"), .default = term)) %>% 
  select(-sex) %>% 
  gt(rowname_col = "term") %>%
  tab_header(
    title = "Summary of GLM models for the VP1 motif in ENDIA"
  ) %>%
  cols_label(
    term = "Predictor",
    RR = "RR",
    `CI lower` = "CI Lower",
    `CI upper` = "CI Upper",
    p.value = "Pr(>|z|)"
  ) %>%
  fmt_number(
    columns = c(RR, `CI lower`, `CI upper`, p.value),
    decimals = 3
  ) %>%
  cols_align(
    align = "left",
    columns = everything()
  ) %>% 
  tab_row_group(
    label = "ENDIA females (Case n = 22, Control n = 49)",
    rows = 1:6
  ) %>% 
  tab_row_group(
    label = "ENDIA males (Case n = 22, Control n = 48)",
    rows = 7:12
  ) %>% 
  tab_style(
    style = cell_fill(color = "lightgrey"),
    locations = cells_row_groups(groups = c("ENDIA males (Case n = 22, Control n = 48)", "ENDIA females (Case n = 22, Control n = 49)"))
  )

as_raw_html(endia_vp1_sex_s_models_tidy_table)
```

```{r, echo = FALSE}
gtsave(endia_vp1_sex_s_models_tidy_table, "figures/endia_vp1_sex_s_models_tidy_table.png")
```

**VIGR** 

```{r, echo = FALSE}
vigr_vp1_xx_model_tidy <- vigr_vp1_xx_model %>% 
  broom.mixed::tidy(effect = "fixed", conf.int = TRUE, exponentiate = TRUE) %>% 
    add_column(sex = "Females")

vigr_vp1_xy_model_tidy <- vigr_vp1_xy_model %>% 
  broom.mixed::tidy(effect = "fixed", conf.int = TRUE, exponentiate = TRUE) %>% 
    add_column(sex = "Males")

vigr_vp1_sex_s_models_tidy <- rbind(vigr_vp1_xx_model_tidy, vigr_vp1_xy_model_tidy) %>% 
    dplyr::rename(RR = estimate,
               `CI lower` = conf.low,
               `CI upper` = conf.high) %>% 
    select(term, RR, `CI lower`, `CI upper`, p.value, sex) %>% 
    filter(term != "(Intercept)")

vigr_vp1_sex_s_models_tidy_table <- vigr_vp1_sex_s_models_tidy %>%
  mutate(term = case_match(term, !!!c("log_total_rpk" ~ "log(EV)"), .default = term)) %>% 
  select(-sex) %>% 
  gt(rowname_col = "term") %>%
  tab_header(
    title = "Summary of GLM models for the VP1 motif in VIGR"
  ) %>%
  cols_label(
    term = "Predictor",
    RR = "RR",
    `CI lower` = "CI Lower",
    `CI upper` = "CI Upper",
    p.value = "Pr(>|z|)"
  ) %>%
  fmt_number(
    columns = c(RR, `CI lower`, `CI upper`, p.value),
    decimals = 3
  ) %>%
  cols_align(
    align = "left",
    columns = everything()
  ) %>% 
  tab_row_group(
    label = "VIGR females (Case n = 10, Control n = 10)",
    rows = 1:4
  ) %>% 
  tab_row_group(
    label = "VIGR males (Case n = 11, Control n = 11)",
    rows = 5:9
  ) %>% 
  tab_style(
    style = cell_fill(color = "lightgrey"),
    locations = cells_row_groups(groups = c("VIGR males (Case n = 11, Control n = 11)", "VIGR females (Case n = 10, Control n = 10)"))
  )

as_raw_html(vigr_vp1_sex_s_models_tidy_table)
```

```{r, echo = FALSE}
gtsave(vigr_vp1_sex_s_models_tidy_table, "figures/vigr_vp1_sex_s_models_tidy_table.png")
```

#### 3D models, including both `case ~ log(EV_normalised_antibody_levels)` and `EV_antibody_levels ~ case`

**ENDIA**

```{r, echo = FALSE}
endia_3d_model_tidy <- endia_3d_model %>% 
  broom.mixed::tidy(effect = "fixed", conf.int = TRUE, exponentiate = TRUE) %>% 
    add_column(outcome = "case")

endia_3d_abundance_as_outcome_model_tidy <- endia_3d_abundance_as_outcome %>% 
  broom.mixed::tidy(effect = "fixed", conf.int = TRUE, exponentiate = TRUE) %>% 
    add_column(outcome = "abundance")

endia_3d_model_tidy_models_tidy <- rbind(endia_3d_model_tidy, endia_3d_abundance_as_outcome_model_tidy) %>% 
    dplyr::rename(RR = estimate,
               `CI lower` = conf.low,
               `CI upper` = conf.high) %>% 
    select(term, RR, `CI lower`, `CI upper`, p.value, outcome) %>% 
    filter(term != "(Intercept)")

endia_3d_model_tidy_models_tidy_table <- endia_3d_model_tidy_models_tidy %>%
  mutate(term = case_match(term, !!!c("log_total_rpk" ~ "log(EV)"), .default = term)) %>% 
  select(-outcome) %>% 
  gt(rowname_col = "term") %>%
  tab_header(
    title = "Summary of GLM models for the 3D motif in ENDIA"
  ) %>%
  cols_label(
    term = "Predictor",
    RR = "RR",
    `CI lower` = "CI Lower",
    `CI upper` = "CI Upper",
    p.value = "Pr(>|z|)"
  ) %>%
  fmt_number(
    columns = c(RR, `CI lower`, `CI upper`, p.value),
    decimals = 3
  ) %>%
  cols_align(
    align = "left",
    columns = everything()
  ) %>% 
  tab_row_group(
    label = "ENDIA case as outcome variable",
    rows = 1:7
  ) %>% 
  tab_row_group(
    label = "ENDIA EV antibody response as outcome variable",
    rows = 8:14
  ) %>% 
  tab_style(
    style = cell_fill(color = "lightgrey"),
    locations = cells_row_groups(groups = c("ENDIA case as outcome variable", "ENDIA EV antibody response as outcome variable"))
  )

as_raw_html(endia_3d_model_tidy_models_tidy_table)
```

```{r, echo = FALSE}
gtsave(endia_3d_model_tidy_models_tidy_table, "figures/endia_3d_model_tidy_models_tidy_table.png")
```

**VIGR**

```{r, echo = FALSE}
vigr_3d_model_tidy <- vigr_3d_model %>% 
  broom.mixed::tidy(effect = "fixed", conf.int = TRUE, exponentiate = TRUE) %>% 
  add_column(outcome = "case")

vigr_3d_abundance_as_outcome_tidy <- vigr_3d_abundance_as_outcome %>% 
  broom.mixed::tidy(effect = "fixed", conf.int = TRUE, exponentiate = TRUE) %>% 
    add_column(outcome = "abundance")

vigr_3d_model_tidy_models_tidy <- rbind(vigr_3d_model_tidy, vigr_3d_abundance_as_outcome_tidy) %>% 
    dplyr::rename(RR = estimate,
               `CI lower` = conf.low,
               `CI upper` = conf.high) %>% 
    select(term, RR, `CI lower`, `CI upper`, p.value, outcome) %>% 
    filter(term != "(Intercept)")

vigr_3d_model_tidy_models_tidy_table <- vigr_3d_model_tidy_models_tidy %>%
  mutate(term = case_match(term, !!!c("log_total_rpk" ~ "log(EV)"), .default = term)) %>% 
  select(-outcome) %>% 
  gt(rowname_col = "term") %>%
  tab_header(
    title = "Summary of GLM models for the 3D motif in VIGR"
  ) %>%
  cols_label(
    term = "Predictor",
    RR = "RR",
    `CI lower` = "CI Lower",
    `CI upper` = "CI Upper",
    p.value = "Pr(>|z|)"
  ) %>%
  fmt_number(
    columns = c(RR, `CI lower`, `CI upper`, p.value),
    decimals = 3
  ) %>%
  cols_align(
    align = "left",
    columns = everything()
  ) %>% 
  tab_row_group(
    label = "VIGR case as outcome variable",
    rows = 1:6
  ) %>% 
  tab_row_group(
    label = "VIGR EV antibody response as outcome variable",
    rows = 7:12
  ) %>% 
  tab_style(
    style = cell_fill(color = "lightgrey"),
    locations = cells_row_groups(groups = c("VIGR case as outcome variable", "VIGR EV antibody response as outcome variable"))
  )

as_raw_html(vigr_3d_model_tidy_models_tidy_table)
```

```{r, echo = FALSE}
gtsave(vigr_3d_model_tidy_models_tidy_table, "figures/vigr_3d_model_tidy_models_tidy_table.png")
```

#### Histogram of 3D motif total counts for the antibody levels as outcome models

A = ENDIA
B = VIGR

```{r}
endia_3d_total_counts_distribution_plot + 
  (vigr_3d_total_counts_distribution_plot + 
     theme(axis.title.y = element_blank())) + 
  plot_annotation(tag_levels = "A")
```

```{r, echo = FALSE}
ggsave("figures/3d_motif_distribution.jpg", dpi = 300, width = 7, height = 4)
```

**Supplementary Fig. 7:** Histogram showing the distribution of the total counts for the 3D motif in ENDIA (A) and VIGR (B).  Total counts represent the non-normalised total antibody counts assigned to the 3D region of the Enterovirus genome polyprotein.
